theano/compile/mode.py

```diff
@@ -111,6 +113,16 @@ class AddDestroyHandler(gof.Optimizer):
 
     """
     def apply(self, fgraph):
+        supervisor_added = False
+        for feature in fgraph._features:
+            if isinstance(feature, Supervisor):
+                supervisor_added = True
+                return
+        if not supervisor_added:
+            warnings.warn("WARNING: Supervisor is not added. Please do not"
+                          "use some_op.optimize(fgraph). Instead use theano.function"
+                          "to add the optimiztions.",
```

Comment:
The message isn't good. I think what we need to tell is something like:

WARNING: Supervisor is not added. Please build a FunctionGraph via theano.compile.function_module.std_graph() or add the Supervisor class manually.